pymc/distributions/multivariate.py

```diff
@@ -2192,32 +2192,25 @@ def make_node(self, rng, size, dtype, alpha, K):
         alpha = at.as_tensor_variable(alpha)
         K = at.as_tensor_variable(intX(K))
 
-        if alpha.ndim > 0:
-            raise ValueError("The concentration parameter needs to be a scalar.")
-
         if K.ndim > 0:
             raise ValueError("K must be a scalar.")
 
         return super().make_node(rng, size, dtype, alpha, K)
 
-    def _infer_shape(self, size, dist_params, param_shapes=None):
-        alpha, K = dist_params
-
-        size = tuple(size)
-
-        return size + (K + 1,)
+    def _supp_shape_from_params(self, dist_params, **kwargs):
+        K = dist_params[1]
+        return (K + 1,)
 
     @classmethod
     def rng_fn(cls, rng, alpha, K, size):
         if K < 0:
             raise ValueError("K needs to be positive.")
 
         if size is None:
-            size = (K,)
-        elif isinstance(size, int):
-            size = (size,) + (K,)
+            size = alpha.shape + (K,)
+            alpha = alpha[..., np.newaxis]
```

Comment:
Don't you need this regardless of whether size is provided or not?

Reply:
I think you need to expand always because you are feeding it into beta with (k+1) draws to the right and you want alpha to broadcast to those.

Reply:
yes got it